src/ucp/proto/proto_common.c

```diff
@@ -449,6 +449,30 @@ ucp_proto_common_get_lane_perf(const ucp_proto_common_init_params_t *params,
     return status;
 }
 
+static inline ucp_lane_index_t
+ucp_proto_common_lanes_iter(const ucp_ep_config_key_t *ep_config_key,
```

Comment:
Currently we iterate `lanes` array, that is sorted array of different lanes (in the order added in `ucp_wireup_search_lanes`): [CM, RMA, AMO, AM, TAG, RMA_BW, ...]
When it comes to RMA_BW lanes selection, I see that first lane in the lanes array is always AM lane, which is chosen by latency criteria, so it's not the lane with max BW.
Therefore for RNDV we wrongly select the first lane which is typically AM lane - and this selection is not optimal (check rndv_1_lane test in https://github.com/openucx/ucx/pull/10369).

This is not good, we should normally start selecting from the lanes with highest BW. And that was the reason why we introduced PR to sort lanes by BW, that I mentioned in desc.
But fortunately in ep config we already have a `rma_bw_lanes` array (permutation of lanes indexes), which is already ordered array of only RMA_BW lanes. So the idea was to iterate directly that array to avoid collisions with generic `lanes` array.

Reply:
ok if I get it, previously first RMA lane selected would be because of highest score and might also be AM lane. This highest score might not always mean highest bandwidth. Now for RMA we will use `rma_bw_lanes` which is sorted by bandwidth.

Reply:
`rma_bw_lanes` is also sorted by score, not just by BW. E.g. if there are multiple paths of the same device, they are sorted according to score.
So the most important property of this array is that it contains only RMA_BW lanes without preceding AM lane

Reply:
thanks, after discussing, i understand it is not only sorted by bw but also by path0/path1 etc.